Content.Shared/Temperature/Systems/SharedTemperatureSystem.cs

```diff
@@ -172,8 +290,24 @@ private void OnTemperatureChangeAttempt(Entity<TemperatureProtectionComponent> e
     }
 }
 
+/// <summary>
+/// Event raised whenever the temperature of an entity changes.
+/// </summary>
 [ByRefEvent]
 public readonly record struct OnTemperatureChangeEvent(float CurrentTemperature, float LastTemperature, Entity<TemperatureComponent> Entity)
 {
+    /// <summary>
+    /// The difference between the new temperature and the previous temperature.
+    /// </summary>
     public readonly float TemperatureDelta = CurrentTemperature - LastTemperature;
 }
+
+/// <summary>
+/// <para>Event raised whenever the heat capacity of an entity is recalculated.</para>
+/// <para>Used to accumulate heat capacity from sources such as fixtures and material composition.</para>
+/// </summary>
+[ByRefEvent]
+public record struct HeatCapacityUpdateEvent(Entity<TemperatureComponent> Entity, float HeatCapacity = 0f)
+{
+    public readonly Entity<TemperatureComponent> Entity = Entity;
+}
```

Comment:
See energy conservation issues above.

Reply:
This I see your point on this one. Unfortunately, I don't think there's a general way to handle conservation of thermal energy through the event, especially if you have multiple things modifying heat capacity and only some of them care about conservation. I'll think about it and see if anything comes to mind.

Reply:
Conserving energy is a pretty important requirement. What exactly is the use case for modifying heat capacity in an event? This is something that is quite fundamental to an object (like mass) that shouldn't really change without good reason.

Reply:
In order to prevent float errors from making the heat capacity drift we need to completely rebuild the heat capacity when it's modified instead of just multiplying/adding/subtracting it.

Reply:
Yes, what I mean is, what is the end-use case for multiplying/adding/subtracting heat capacity?

Reply:
Mostly solutions/reagent. Every time you add reagents to or remove reagents from a solution it's heat capacity changes. Currently it uses its own temperature system but that's going to get axed and it's just going to use the standard entity temperature system. As a result we need this system to handle changing heat capacities.